### ISSUE TITLE
LanguageServerPlugin stops Code Completion from working

### ISSUE THREAD

**Issue by yunkot** [eranif/codelite#2608]
I am using CodeLite built from latest revision 2b75baba84d92218ae96fe372497c9ebdccec4bf (November 8, 2020) on Arch Linux, Gnome 3.38.1 on Wayland, kernel 5.9.8-arch1-1. The issue can also be reproduced on Ubuntu 20.04.1 LTS, Gnome 3.36.3 on X11, kernel 5.4.0-53-generic, CodeLite 14.0.2 (don't know exact revision as I've lost the source git folder). 

On Arch Linux, I am using CodeLite with packaged wxWidgets, built with  "-DWITH_WX_CONFIG=wx-config-gtk3" option in CMAKE, whereas on Ubuntu I am using wxWidgets built according to Wiki (from "Using GTK3" section).

Steps to reproduce:

1. Remove ".codelite" configuration folder, e.g. "rm -r -f ~/.codelite"
2. Start CodeLite.
3. In Startup Dialog, select Next, Default Profile, Scan (for compilers), then pick Dark Theme instead of "System Default", click Next, then Finish. CodeLite will restart.
4. Create new C++ WorkSpace and new Console C++ Project with GCC as compiler and default settings.

In resulting "main.cpp", create some test function:

```
void testFunc(int a, int b)
{
}
```

Then, inside "int main(int argc, char **argv)" function, type "testF" and try invoking auto-complete (Ctrl+Space). 
Expected: a suggestion for "testFunc". 
Actual: nothing happens. 
Ctrl+Enter does work, however. 

It looks like for some reason auto-completion tooltip window cannot be shown. Same thing happens if you type "testFunc(" - there is no tooltip for its parameters shown, even if you press Ctrl+Shift+Space. 

If you repeat the above steps, but instead of choosing "Dark Theme", use "System Default" (which looks similar if you are using dark theme anyway), then auto-completion appears to work fine.

**Comment by eranif** [comment]
I can't even think of a reason how to two are  related ...
I am using CodeLite on Ubuntu / CentOS with dark theme and I never encountered this. Does this also happens to you on Ubuntu or just on Arch?

**Comment by yunkot** [comment]
Yes, as I've written above, I could reproduce this equally on Arch and Ubuntu 20.04.1 LTS. Since original report in November, on all installations, I had to pick "System Default" to allow code completion to work in CodeLite, which I'm always compiling from GitHub. Please let me know if you need more information from my side to determine the cause of the problem.

**Comment by yunkot** [comment]
I am not sure if it is related, but also on my configuration when debugging and placing mouse cursor over variables, the popup window that shows variable values, appears close to top/left of the display, very far from the variable position in the source code or   mouse position at that moment. I am using multi-monitor configuration, where left monitor is in portrait mode, and CodeLite runs on that monitor. 

However, in case of code completion when using "Dark Theme", it doesn't work completely, so it's not like the window is somewhere off the screen. If you think this is not related, I can report it as a separate issue.

![Screenshot from 2020-12-17 17-00-39](https://user-images.githubusercontent.com/14065837/102554061-a18cf980-4089-11eb-9319-27862081925f.png)

**Comment by eranif** [comment]
which version of wxWidgets are you using?

**Comment by yunkot** [comment]
On Arch, I am using wxgtk3 3.0.5.1-1 that is pre-packaged (this is where I am getting above issue with debug popup position being very far from the actual location). On Ubuntu, I am using it compiled for GTK3, from GitHub.

**Comment by eranif** [comment]
Can you build the latest wxWidgets and see if this issue persist?

**Comment by yunkot** [comment]
I've built latest CodeLite from latest revision 0e12e2d1710dec9a51780a53dedbed4acdb6540c along with wxWidgets from [latest revision](https://github.com/wxWidgets/wxWidgets/commit/1324b5122baf540e61796daf8f803408856afb76) on Ubuntu 20.04.1 LTS 64-bit with kernel 5.4.0-58-generic. The issue can still be reproduced exactly as described in original post.

**Comment by yunkot** [comment]
This issue can also be reproduced in CodeLite 14.0.4 on Windows and is quite annoying one as you are really stuck with default white theme if you want to keep using Code Completion.
Once you pick anything other than System Default theme, code completion stops working and there is no way to make it work even if you run Setup Wizard again and pick Default theme. To make it work again, you have to remove CodeLite configuration folder from AppData\Roaming, then run CodeLite and pick Default theme.

However, while code completion is working under Default theme, if you run Setup Wizard to change it to Dark Theme (that is, without removing configuration folder), code completion stops working.

In other words, changing to anything from Default theme disables code completion, but changing back to Default theme does not make it work again unless you remove the whole configuration folder and pick Default theme after restarting CodeLite during first time setup wizard.

**Comment by yunkot** [comment]
After aforementioned experiments, there are two cases - CodeLite using Default theme that has code completion working and CodeLite using Default theme that has it non-working (because initially Dark theme was chosen and Setup Wizard was executed again to pick Default theme, which doesn't fix the problem unless you remove configuration folder).

So, with both configurations seemingly the same, I was comparing the configuration folders between working and non-working configurations. It seems that when you first run CodeLite and use Default theme, it does not enable any of the plugins. This is the main difference between chosing Default theme and any other theme option during first startup. I think this might be a bug by itself, but it's an unrelated issue.

Please note that I am using default/fresh install of CodeLite, so it's only using plugins that are bundled in the installation. So went disabling plugins that are enabled by default one by one until code completion started to work again. 

The plugin that stops Code Completion from working is "LanguageServerPlugin". Disabling it makes code completion work again. The reason why code completion was working in "Default" theme settings is because when you pick that during first CodeLite start, it doesn't enable any of the plugins and thus "LanguageServerPlugin" isn't enabled in that case.

So as a workaround until the problem gets fixed is to disable "LanguageServerPlugin".
